monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -96,26 +98,38 @@ def initialize(
 
     # This should pull from Simulation entity
     @classmethod
-    def get_first_monkey_time(cls):
-        agents = cls._agent_repository.get_agents()
-
-        return min(agents, key=lambda a: a.start_time).start_time
+    def get_first_monkey_time(cls) -> Optional[datetime]:
+        try:
+            agents = cls._agent_repository.get_agents()
+            return min(agents, key=lambda a: a.start_time).start_time
+        except (RetrievalError, ValueError) as err:
+            logger.error(f"Error occurred while getting first monkey time: {err}")
+            return None
 
     @classmethod
-    def get_last_monkey_dead_time(cls):
-        agents = filter(lambda a: a.stop_time is not None, cls._agent_repository.get_agents())
-
-        return max(agents, key=lambda a: a.stop_time).stop_time
+    def get_last_monkey_dead_time(cls) -> Optional[datetime]:
+        try:
+            agents = filter(lambda a: a.stop_time is not None, cls._agent_repository.get_agents())
+            return max(agents, key=lambda a: a.stop_time).stop_time
+        except (RetrievalError, ValueError) as err:
+            logger.error(f"Error occurred while getting last monkey stop time: {err}")
+            return None
 
     @staticmethod
-    def get_monkey_duration():
-        delta = ReportService.get_last_monkey_dead_time() - ReportService.get_first_monkey_time()
+    def get_monkey_duration() -> str:
         st = ""
-        hours, rem = divmod(delta.seconds, 60 * 60)
-        minutes, seconds = divmod(rem, 60)
+        try:
+            delta = (
+                ReportService.get_last_monkey_dead_time() - ReportService.get_first_monkey_time()
+            )
+            hours, rem = divmod(delta.seconds, 60 * 60)
+            minutes, seconds = divmod(rem, 60)
+
+            if delta.days > 0:
+                st += "%d days, " % delta.days
+        except TypeError:
+            hours, minutes, seconds = (0, 0, 0)
 
-        if delta.days > 0:
```

Comment:
Why is this removed?

Reply:
It is not 
<img width="703" alt="image" src="https://user-images.githubusercontent.com/15820737/200304580-6c5face8-6149-40ee-9613-4e58616d8ffd.png">